ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -12,82 +12,97 @@ export const ApplicationResourceList = ({
     onNodeClick,
     nodeMenu
 }: {
-    resources: models.ResourceStatus[];
+    resources: any[];
     onNodeClick?: (fullName: string) => any;
     nodeMenu?: (node: models.ResourceNode) => React.ReactNode;
-}) => (
-    <div className='argo-table-list argo-table-list--clickable'>
-        <div className='argo-table-list__head'>
-            <div className='row'>
-                <div className='columns small-1 xxxlarge-1' />
-                <div className='columns small-2 xxxlarge-2'>NAME</div>
-                <div className='columns small-3 xxxlarge-4'>GROUP/KIND</div>
-                <div className='columns small-3 xxxlarge-3'>NAMESPACE</div>
-                <div className='columns small-2 xxxlarge-2'>STATUS</div>
+}) => {
+    const [showSyncOrder, setShowSyncOrder] = React.useState(false);
+    React.useEffect(() => {
+        resources.forEach(resource => {
+            if (typeof resource.syncOrder !== 'undefined' && resource.syncOrder?.length !== 0) {
+                setShowSyncOrder(true);
```

Comment:
Once we set this to true, can we break out of the `forEach` to avoid doing extra work?

Reply:
Good catch! Thanks for the review. Code updated as per comment.